web-server/models/prosumer.js

```diff
@@ -0,0 +1,155 @@
+        this.house_filename = data.house_filename || null;
+        this.time = data.time || new Date();
+    }
+    
+    
+    /**
+     * Tries to only find one prosumer in the database.
+     * @returns {Promise} that resolves to only one if successful
+     */
+    static findOne(where) {
+        return new Promise((resolve, reject) => {
+            (async () => {
+                try {
+                    let prosumers = await Prosumer.findMany(where);
+                    if (prosumers.length != 1) {
+                        reject("did not find one prosumer but " + prosumers.length + " instead");
+                    } else {
+                        resolve(prosumers[0]);
+                    }
+                } catch (err) {
+                    reject(err);
+                }
+            })()
+        });
+    }
+
+
+    /**
+     * Tries to find many prosumers in the database.
+     * @param {Object} where key-value pairs to match agains
+     * @returns {Promise} that resolves to a list of prosumers if successful
+     */
+    static findMany(where) {
+        return new Promise((resolve, reject) => {
+            (async () => {
+                try {
+                    var result = [];
+                    let { rows } = await db.select('prosumers', where);
+                    for (var i = 0; i < rows.length; i++) {
+                        let row = rows[i];
+                        let user = await db.select('users', {id: row.id});
+                        if (user.rows.length == 1) {
+                            const data = {...row, ...user.rows[0]};
+                            result.push(new Prosumer(data));
+                        }
+                    }
+                    resolve(result);
+                } catch (err) {
+                    reject(err);
+                }
+            })();
+        });
+    }
+
+
+    /**
+     * Stores this prosumer in the database tables users and prosumers
+     * with a password.
+     * Note: you DO need to hash the password before calling this.
+     */
+    store() {
+        (async () => {
+            await super.store();
+            const queryText = `INSERT INTO prosumers(id, time, production, consumption,
+                               buffer, buffer_max, buffer_storing_limit, house_filename)
+                               VALUES($1, $2, $3, $4, $5, $6, $7, $8)`;
+            const params = [
+                this.id, this.time, this.production, this.consumption, this.buffer,
+                this.buffer_max, this.buffer_storing_limit, this.house_filename];
+            await db.query(queryText, params);
+        })();
+    }
+
+    
+
+    /**
+     * Updates the specified fields from the prosumer/user object in the database.
+     * Note: updated_at is automatically updated in the users  with the current time.
+     * @param {array} fields containing strings of each field to include.
+     */
+    update(fields) {
```

Comment:
Why have you added this function? I don't see the need to update the prosumers historical data.

Reply:
The only field that needs to updated is the house_filename, but I repurposed the prosumer_data to hold current information about the prosumer not historical data. This is so the simulator also can restart and use those values to setup e.g. battery information. But production and consumption is per step basis so this should not be stored here. Do you need the prosumer_data table? It was not used anywhere in the project.